[WebCore] Remove old libxslt code smell

LumioseSil opened the pull request:
#### b6426694e8cecf14aa9c80b96509745bf07cc876
<pre>
[WebCore] Remove old libxslt code smell
<a href="https://bugs.webkit.org/show_bug.cgi?id=274573">https://bugs.webkit.org/show_bug.cgi?id=274573</a>

Reviewed by Darin Adler.

libxslt 1.1.13 shipped over a decade ago. We can
remove this code smell now.

* Source/WebCore/xml/XSLTProcessorLibxslt.cpp:
 (WebCore::XSLTProcessor::transformToString):
 Remove old workaround.
* Source/cmake/OptionsGTK.cmake: Set libxslt to
  1.1.13.
* Source/cmake/OptionsMac.cmake: Ditto.
* Source/cmake/OptionsWPE.cmake: Ditto.

Canonical link: <a href="https://commits.webkit.org/282807@main">https://commits.webkit.org/282807@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/12e18eadb92de03b3d307645d80617a047d98c85

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/64319 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43676 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16916 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/68341 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14927 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/66439 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/51374 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/15207 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51762 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10297 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/67388 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40379 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55651 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32381 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37049 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/13030 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13801 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/57431 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59015 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/13359 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70040 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/63564 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/8266 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12878 "Found 1 new test failure: media/track/track-text-track-destructor-crash.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59083 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/8299 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55742 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/59257 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14201 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6833 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/517 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/85325 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/39496 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15049 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/40575 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/41758 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/40318 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (changes requested):
From the point of view of the GTK and WPE ports, you would also need to change `Source/cmake/Options{GTK,WPE}.cmake` to check that the libxslt found is version 1.1.13 or newer&mdash;currently we check for 1.1.7 there.

darinadler commented on Source/cmake/OptionsGTK.cmake:
```diff
@@ -303,7 +303,7 @@ if (ENABLE_GAMEPAD)
 
 if (ENABLE_XSLT)
-    find_package(LibXslt 1.1.7 REQUIRED)
+    find_package(LibXslt 1.1.42 REQUIRED)
```
EWS seems to have 1.1.38

darinadler commented on Source/cmake/OptionsMac.cmake:
```diff
@@ -88,4 +88,4 @@ set(USE_ANGLE_EGL ON)
 find_package(ICU 61.2 REQUIRED COMPONENTS data i18n uc)
 find_package(LibXml2 2.8.0 REQUIRED)
-find_package(LibXslt 1.1.7 REQUIRED)
+find_package(LibXslt 1.1.42 REQUIRED)
```
My macOS Sequoia system seems to have 1.1.35

darinadler reviewed (changes requested):
Need to figure out the right minimum version. Maybe `1.1.13` is the way to go for now?

aperezdc reviewed (commented):
Patch LGTM from the POV of the GTK and WPE ports now that CMake checks for version `1.1.13`.

The Windows port was already checking for a newer version so I assume there won't be any problem on their end (cc / @donny-dont).

I'll leave the final patch approval to the Apple folks.

darinadler reviewed (approved):
It’s a bit peculiar to say “code smell” here. I’m honestly not even sure it’s precisely the right term for “remove [harmless] workaround we no longer need”. We’d typically reserve that term for something more consequential if we’d use it at all.

LumioseSil committed:
[WebCore] Remove old libxslt code smell
https://bugs.webkit.org/show_bug.cgi?id=274573

Reviewed by Darin Adler.

libxslt 1.1.13 shipped over a decade ago. We can
remove this code smell now.

* Source/WebCore/xml/XSLTProcessorLibxslt.cpp:
 (WebCore::XSLTProcessor::transformToString):
 Remove old workaround.
* Source/cmake/OptionsGTK.cmake: Set libxslt to
  1.1.13.
* Source/cmake/OptionsMac.cmake: Ditto.
* Source/cmake/OptionsWPE.cmake: Ditto.

Canonical link: https://commits.webkit.org/282807@main
